### PR TITLE
chore: dynamic project for ExampleSuite compilation

### PR DESCRIPTION
**Pull request by octonato**
Inspired by #795, but able to load the example suite projects dynamically using flags:

```
sbt -Dexample.suite.scala.enabled -Dexample.suite.java.enabled
```

I found this quite useful when working on #796. As such as could generate the code and compile without leaving sbt or switching consoles.

### CODE REVIEW

**Comment by octonato** [comment]
Quite strange, the ExampleSutie for Scala failed to find the generated the classes 

```scala
==> X com.akkaserverless.codegen.scalasdk.ExampleSuite.event-sourced-entity / domain-in-service-package / unmanaged  0.005s munit.FailException: /home/circleci/project/codegen/core/src/test/scala/com/lightbend/akkasls/codegen/ExampleSuiteBase.scala:109 Files should have been generated but are missing: [org/example/eventsourcedentity/Counter.scala]
108:
109:    assert(missing.isEmpty, s"Files should have been generated but are missing: [${missing.mkString(", ")}]")
110:    assert(extra.isEmpty, s"Unexpected files were generated: [${extra.mkString(", ")}]")
    at munit.FunSuite.assert(FunSuite.scala:11)
    at com.lightbend.akkasls.codegen.ExampleSuiteBase.assertFiles(ExampleSuiteBase.scala:109)
    at com.lightbend.akkasls.codegen.ExampleSuiteBase.$anonfun$new$4(ExampleSuiteBase.scala:79)
```

Obviously, it works on my machine.

**Comment by octonato** [comment]
Ok, I understand how this got broken and it means that the main branch is broken for all PRs now. 

This is happening because both #791 and #793 where merged in parallel. We should have rebased #791 with #793.

In #793, we added ExampleSuite tests for Scala. In #791, we fixed the issue with the package issue (we prioritize javaPackage over the protoPackage). The problem is that in Scala we get an extra package that uses the name of proto file.

For instance, the `Counter.scala` file is now being generated in `org.example.eventsourcedentity.counter_api` when previously it was `org.example.eventsourcedentity`. And now the ExampleSuite tests for Scala is failing. 

The good news is that it show how useful the ExampleSuite is useful for us. 

Anyway, I will fix this tomorrow afternoon.

**Comment by octonato** [comment]
Back to draft because now on top of #803

File: build.sbt [hunk 1]
```diff
@@ -331,6 +331,54 @@ lazy val codegenScalaCompilationTest = project
...
+          .disablePlugins(HeaderPlugin, ScalafmtPlugin)
+          .settings(
+            Compile / unmanagedSourceDirectories ++= Seq("generated-managed", "generated-unmanaged").map(
+              baseDirectory.value / _),
+            Test / unmanagedSourceDirectories ++= Seq("generated-test-managed", "generated-test-unmanaged").map(
+              baseDirectory.value / _),
+            Compile / PB.protoSources += baseDirectory.value / "proto",
+            akkaGrpcGeneratedLanguages := Seq(AkkaGrpc.Scala),
+            akkaGrpcCodeGeneratorSettings += "flat_package",
+            libraryDependencies ++= Seq(Dependencies.akkaslsSdkProtocol % "protobuf-src"))
```
Comment (octonato) [inline build.sbt:374]: Not clear to me why we need this two lines, but that's how I made it compiling the files correctly. 🤷

**Review by jrudolph — APPROVED** [review]
Nice! I put in the comments last week but forget to post them :facepalm:

**Comment by octonato** [comment]
@jrudolph, I will merge as is for now because I want need this to move fast in my other PR #796. 

I had experimented with a method receiving a `Project => Project` but I wasn't very happy. I will give it another try on a follow-up PR.

**Pull request merged**